src/com/yetanalytics/lrs_admin_ui/functions/reaction.cljs

```diff
@@ -82,3 +83,30 @@
   (if (and (= [:ruleset :conditions] [seg0 seg1]) seg2)
     (into [seg0 seg1 seg2 1] rest-seg)
     path))
+
+(defn validate-template-xapi
+  "Take raw JSON str of an xAPI Statement and issue a vec of error maps for 
+   any schema violations."
+  [raw-json]
+  (try
+    (let [result (xs/validate-statement-data raw-json)]
+      result)
+    ;; JSON errors handled by editor directly, ignore
+    (catch js/SyntaxError e)
+    ;; Other exceptions will be spec errors
+    (catch js/Error e
+      (reduce
+       (fn [agg {:keys [val path]}]
+         (cond-> agg
+           (not
+            (and (map? val)
+                 (seq val)
+                 (-> val keys first name (= "$templatePath"))))
```

Comment:
For future-proofing it might be better to check the map for the key rather than seq-ing it

Reply:
went back and resolved